### ISSUE TITLE
Latest docker update causes issues on btrfs

### ISSUE THREAD

**Issue by kernle32dll** [void-linux/void-packages#19202]
The commit https://github.com/void-linux/void-packages/commit/ec709f9e09458bb1707a4136c1d4635ca11b0e7a did not only update docker, but enforced the `overlay2` storage driver. While this is a fine default, it can actually break docker on `btrfs` file systems.

From the [Dockerd documentation](https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-storage-driver):

> Note: Both overlay and overlay2 are currently unsupported on btrfs or any Copy on Write filesystem and should only be used over ext4 partitions.

Even if not breaking, its an un-intuitive change, if you don't happen to use `overlay2` previously. It took me a good while to understand why all of my images and volumes were gone - which was due to the change of the storage driver.

Its easy to fix locally - just remove the `storage-driver` line from `/etc/docker/daemon.json`. But this should be removed in the package, too - IMO.

**Comment by D-Nice** [comment]
I find many issues with the change myself. The only potentially needed part there is log-driver, which should just be passed as a flag from runit, and not bog the rest of us down with some defaults we don't want AND replacing our own configs.

**Comment by xtraeme** [comment]
@jbenden

**Comment by kernle32dll** [comment]
Welp, the new PR now completely broke Docker for me. Daemon is not starting again anymore.

`sv status docker` yields:

`run: docker: (pid 15449) 1s; warning: docker/log: unable to open supervise/ok: file does not exist`

No clue how to fix myself.

**Comment by xtraeme** [comment]
Has been fixed this morning, make sure your system is up to date

**Comment by kernle32dll** [comment]
It is. After some tinkering I got it running again:

First, I removed the `/etc/sv/docker/log` folder - not sure if that did anything, but it got rid of the error message above.

Then turns out, due to the original problem I had both `overlay2` and `btrfs` under `/var/lib/docker`. With the removal of the daemon.json per https://github.com/void-linux/void-packages/pull/19226, Docker was unable to "guess" the correct storage driver. I deleted the `overlay2` folder (warning fellow Googler, this is due to me using Btrfs, don't do that blindly), and it works again.
